File: workshop_content/KI_WORKSHOP/C#_Vergleichsgruppe/Provisionierung/Domain/FesterProvisionContext.cs
namespace Provisionierung.Domain;

/// <summary>
/// Für Tests: Feste Zeitkonfiguration.
/// </summary>
public class FesterProvisionContext : IProvisionContext
{
    private readonly DateTime _festesDatum;
    private readonly bool _freitagNach15Uhr;

    public FesterProvisionContext(DateTime festesDatum, bool freitagNach15Uhr = false)
    {
        _festesDatum = festesDatum;
        _freitagNach15Uhr = freitagNach15Uhr;
    }

    public DateTime AktuellesDatum => _festesDatum;

    public bool IstFreitagNach15Uhr() => _freitagNach15Uhr;
}

File: workshop_content/KI_WORKSHOP/Java_Vergleichsgruppe/src/main/java/provisionierung/domain/FesterProvisionContext.java
package provisionierung.domain;

import java.time.LocalDateTime;

/**
 * Für Tests: Feste Zeitkonfiguration.
 */
public class FesterProvisionContext implements ProvisionContext {
    private final LocalDateTime festesDatum;
    private final boolean freitagNach15Uhr;

    public FesterProvisionContext(LocalDateTime festesDatum, boolean freitagNach15Uhr) {
        this.festesDatum = festesDatum;
        this.freitagNach15Uhr = freitagNach15Uhr;
    }

    @Override
    public LocalDateTime getAktuellesDatum() {
        return festesDatum;
    }

    @Override
    public boolean istFreitagNach15Uhr() {
        return freitagNach15Uhr;
    }
}
